.github/workflows/commit.yaml

```diff
@@ -54,19 +57,32 @@ jobs:
       - name: "Checkout"
         uses: actions/checkout@v2
 
-      - name: "Install Go"
-        uses: actions/setup-go@v2
-        with:
-          go-version: ${{ env.GO_VERSION }}
-
-      - name: "Cache golang"
+      # This sets the version-specific GOROOT env on macOS until actions/virtual-environments#4156
+      #
+      # This must happen before interpolating ${!GOROOT_NAME} on macOS
+      - name: "Set GOROOT_X64 env (macOS)"
+        if: runner.os == 'macOS'
+        run: |  # ex GOROOT_1_17_X64 -> 1.17
+          go_version=$(echo ${GOROOT_NAME//_/.}| cut -d. -f2,3)
+          go_root=$(ls -d /Users/runner/hostedtoolcache/go/${go_version}*/x64|sort -n|tail -1)
+          echo "${GOROOT_NAME}=${go_root}" >> $GITHUB_ENV
+
+      - name: "Configure Go"
+        run: |  # This exports variables needed to run and cache a GOROOT set with ${!GOROOT_NAME}.
+          go_path=${!GOROOT_NAME}/bin
+          for var in GOROOT GOCACHE GOMODCACHE; do
+            echo ${var}=$(${go_path}/go env ${var}) >> $GITHUB_ENV
+          done
+          echo ${go_path} >> $GITHUB_PATH
+
+      - name: "Cache Go"
         uses: actions/cache@v2
         with:
-          path: |  # TODO: go build cache if we care, noting it is OS-specific
-            ~/go/pkg/mod
```

Comment:
TODO: It is really only GOCACHE that is OS-dependent. GHA runners don't set any GO related env variables.

So the main thing is that GOCACHE is sadly different on literally OS we use (as defined by go itself)...
* linux - $HOME/.cache/go-build
* macOS - $HOME/Library/Caches/go-build
* windows - %HOME%\AppData\Local\go-build

while GOMODCACHE is luckily always the same ~/go/pkg/mod